File: backend/app/schemas/ai_cost.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AICallCostOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    kind: str
    job_id: int | None
    video_id: int
    provider: str
    model: str
    input_tokens: int | None
    output_tokens: int | None
    latency_ms: int
    cost_usd: float | None
    cost_note: str | None
    confirmed_price: bool
    created_at: datetime


class GroupCostOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    label: str
    total_cost_usd: float
    call_count: int
    unpriced_call_count: int
    all_confirmed: bool


class StoryCostOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    story_job_id: int
    video_id: int
    total_cost_usd: float
    call_count: int
    unpriced_call_count: int


class BatchCostOut(BaseModel):
    batch_id: int
    batch_name: str
    total_cost_usd: float
    story_count: int
    unpriced_call_count: int


class VideoCostOut(BaseModel):
    video_compose_job_id: int
    project_id: int | None
    image_cost_usd: float | None
    image_count: int | None


class AICostSummaryOut(BaseModel):
    total_ai_cost_usd: float
    total_calls: int
    unpriced_call_count: int
    videos_generated: int
    average_cost_per_video_usd: float | None
    average_cost_per_video_note: str
    cost_per_1000_videos_usd: float | None
    by_provider: list[GroupCostOut]
    by_model: list[GroupCostOut]
    by_month: list[GroupCostOut]

File: frontend/src/types/aiCost.ts
// Mirrors backend/app/schemas/ai_cost.py exactly -- see
// docs/features/75-ai-cost-tracking.md.

export interface AICallCost {
  id: number;
  kind: string;
  job_id: number | null;
  video_id: number;
  provider: string;
  model: string;
  input_tokens: number | null;
  output_tokens: number | null;
  latency_ms: number;
  cost_usd: number | null;
  cost_note: string | null;
  confirmed_price: boolean;
  created_at: string;
}

export interface GroupCost {
  label: string;
  total_cost_usd: number;
  call_count: number;
  unpriced_call_count: number;
  all_confirmed: boolean;
}

export interface StoryCost {
  story_job_id: number;
  video_id: number;
  total_cost_usd: number;
  call_count: number;
  unpriced_call_count: number;
}

export interface BatchCost {
  batch_id: number;
  batch_name: string;
  total_cost_usd: number;
  story_count: number;
  unpriced_call_count: number;
}

export interface VideoCost {
  video_compose_job_id: number;
  project_id: number | null;
  image_cost_usd: number | null;
  image_count: number | null;
}

export interface AICostSummary {
  total_ai_cost_usd: number;
  total_calls: number;
  unpriced_call_count: number;
  videos_generated: number;
  average_cost_per_video_usd: number | null;
  average_cost_per_video_note: string;
  cost_per_1000_videos_usd: number | null;
  by_provider: GroupCost[];
  by_model: GroupCost[];
  by_month: GroupCost[];
}
